yaml/metadata.go

```diff
@@ -4,15 +4,25 @@
 
 package yaml
 
-import "github.com/go-vela/types/pipeline"
-
-// Metadata is the yaml representation of
-// the metadata block for a pipeline.
-// nolint:lll // jsonschema will cause long lines
-type Metadata struct {
-	Template bool  `yaml:"template,omitempty" json:"template,omitempty" jsonschema:"description=Enables compiling the pipeline as a template.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-template-tag"`
-	Clone    *bool `yaml:"clone,omitempty" json:"clone,omitempty" jsonschema:"default=true,description=Enables injecting the default clone process.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-clone-tag"`
-}
+import (
+	"github.com/go-vela/types/pipeline"
+)
+
+type (
+	// Metadata is the yaml representation of
+	// the metadata block for a pipeline.
+	// nolint:lll // jsonschema will cause long lines
+	Metadata struct {
+		Template    bool     `yaml:"template,omitempty" json:"template,omitempty" jsonschema:"description=Enables compiling the pipeline as a template.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-template-tag"`
+		Clone       *bool    `yaml:"clone,omitempty" json:"clone,omitempty" jsonschema:"default=true,description=Enables injecting the default clone process.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-clone-tag"`
+		Environment []string `yaml:"environment,omitempty" json:"environment,omitempty" jsonschema:"default=true,description=Controls which containers processes can have global env injected.\nReference: https://go-vela.github.io/docs/reference/yaml/metadata/#the-environment-tag"`
+	}
+
+	// helper type that allows the unmarshaler interface
+	// to add default values back into metadata. Using the
+	// Metadata type directly will result in a reflection error.
+	_metadata Metadata
```

Comment:
What are your thoughts on declaring an anonymous struct instead of this?

In case it's not clear what I'm proposing, I'll drop a comment later on where this type is used 👍

Reply:
Yeah, I can swap around to that! I honestly wasn't sure how to handle it and didn't recognize we had that pattern. Easy change.